### ISSUE TITLE
Integration with Rest Cache

### ISSUE THREAD

**Issue by AlexeyVlasenko** [ComfortablyCoding/strapi-plugin-publisher#19]
Hi!
Thanks for your plugin!

It would be significant enhancement to purge REST Cache (https://strapi-community.github.io/strapi-plugin-rest-cache/) on publish/unpublish

**Comment by ComfortablyCoding** [comment]
I agree, but I don't want to create a direct dependency on another plugin.

I propose instead we add in hooks for when publish and unpublish are called. It will be left up to the user to implement.  This will allow freedom to integrate with any plugin that allows access to it's APIs.

The hooks will be something along the lines of publish before/after and unpublish before/after for the first version. Additional ones can be added as needed.

For example for busting cache for REST cache on unpublish/publish would look have the following sample config.

```js
module.exports = ({ env }) => ({ 	
  publisher: {
    enabled: true, 	
    config:{
      hooks:{
        'publish:after': ( entity, mode) => {
              // Hit REST cache, cache bust endpoint here
         },
         'unpublish:after': ( entity, mode) => {
              // Hit REST cache, cache bust endpoint here
         }
      }
    }
  }
});
```

What do you think?

**Comment by matepaiva** [comment]
I like it :)

**Comment by nogumaruo** [comment]
Hi!
Thanks for the nice plugin!

I would like the validate check to run when content is published. But if I use a plugin to schedule delivery, the validate check doesn't run.
As far as the above is concerned, is using hooks the current workaround?

Why not validate with a plugin?
Am I correct in my understanding that you use hooks?

**Comment by ComfortablyCoding** [comment]
@nogumaruo I am not fully understanding what you are asking. What validate check are you referring to?

**Comment by nogumaruo** [comment]
@ComfortablyCoding 
For example, if the required option for the text component is set to true in "content-type-builder". (available in the advanced settings of each component)
My understanding of the "content-type-builder" behavior is that it does not validate for these when saving. The only time they are validated is when the Publish button is pressed.

However, in the case of a scheduled submission, there are no required validations, so I was wondering if the validations should be performed when a scheduled submission is set up.
